optimum_benchmark/main.py

```diff
@@ -78,11 +77,12 @@ class ExperimentConfig:
     )
 
     # ENVIRONMENT CONFIGURATION
-    environment: DictConfig = DictConfig(
-        {
+    environment: Dict = field(default_factory=lambda: {
             "optimum_version": optimum_version,
             "transformers_version": transformers_version,
+            "transformers_commit": None,
```

Comment:
can't we put here ?
`transformers_commit": get_git_revision_hash(os.getenv("TRANSFORMERS_PATH"))`
or use a resolver ?

Reply:
Unfortunately if the env variable `TRANSFORMERS_PATH` is set in the yml, it is not already visible at this point. An alternative could be to autodetect the repo path and check whether it is a git repo, but it is probably overkill.

Reply:
ah I see. then a resolver would be best.
I prefer not to set the values in the experiment config during the benchmark because even though the manually saved `hydra_config.yaml` will have those changes, the automatically saved `./hydra/config.yaml` won't.